components/brave_vpn/browser/connection/ikev2/win/ras_utils.cc

```diff
@@ -422,6 +422,26 @@ RasOperationResult SetConnectionParamsUsingPowerShell(
   return GetRasSuccessResult();
 }
 
+// `Set-VpnConnectionProxy` cmdlet:
+// https://learn.microsoft.com/en-us/powershell/module/vpnclient/set-vpnconnectionproxy?view=windowsserver2022-ps
+RasOperationResult SetConnectionProxyParamsUsingPowerShell(
+    const std::wstring& entry_name,
+    const std::wstring& pac_file_url) {
+  base::CommandLine power_shell(base::FilePath(L"PowerShell"));
+  power_shell.AppendArg("Set-VpnConnectionProxy");
+  power_shell.AppendArg("-ConnectionName");
+  power_shell.AppendArg(base::WideToUTF8(entry_name));
```

Comment:
we should verify that:

- entry_name follows the regex `[A-Za-z0-9][A-Za-z0-9_-]*`
- pac_file_url is a valid URL

Reply:
this should be done inline, just before instantiating the `base::CommandLine`

Reply:
regex definitely not a good way to do it - but there's a validation method we can use:
https://learn.microsoft.com/en-us/windows/win32/api/ras/nf-ras-rasvalidateentrynamea

Reply:
Done - see latest commit

BTW - we still need to add a profile preference before this work is complete. Setting back to draft

Reply:
shouldn't we at least check it starts with `https://`?

Reply:
Fixed